internal/captain/template.go

```diff
@@ -0,0 +1,93 @@
+package captain
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"text/template"
+	"unicode"
+
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/spf13/cobra"
+)
+
+type Templater struct {
+	Cmd           *Command
+	CommandGroups []CommandGroup
+}
+
+func (t *Templater) RootCmdUsageFunc() func(c *cobra.Command) error {
+	return func(c *cobra.Command) error {
+		tpl := template.New("root_usage_tpl")
+		tpl.Funcs(t.templateFuncs())
+		template.Must(tpl.New("usage_tpl").Parse(localizedTemplate(t.Cmd, "usage_tpl")))
+		template.Must(tpl.Parse(locale.Tt("root_usage_tpl")))
+		return tpl.Execute(os.Stdout, c)
+	}
+}
+
+func (t *Templater) defaultUsageFunc() func(c *cobra.Command) error {
+	return func(c *cobra.Command) error {
+		tpl := template.New("usage_tpl")
+		tpl.Funcs(t.templateFuncs())
+		template.Must(tpl.Parse(localizedTemplate(t.Cmd, "usage_tpl")))
+		return tpl.Execute(os.Stdout, c)
+	}
+}
+
+func (t *Templater) cmdGroupsString(c *cobra.Command) string {
+	var groups []string
+	for _, cmdGroup := range t.cmdGroups(c) {
+		cmds := []string{cmdGroup.Message}
+		for _, cmd := range cmdGroup.Commands {
+			if cmd.cobra.IsAvailableCommand() {
+				cmds = append(cmds, fmt.Sprintf("  %s %s", rpad(cmd.Use(), cmd.cobra.NamePadding()), cmd.Description()))
+			}
+		}
+		groups = append(groups, strings.Join(cmds, "\n"))
+	}
+	return strings.Join(groups, "\n\n")
+}
```

Comment:
Imagine we can loose this function entirely by moving the group registration to the command itself, but some concerns just for future reference;

 - You're hardcoding a lot of templating logic here, that logic should be in the template
 - Try to return/continue early. eg. in your cmd loop invert the conditional and turn it into a `continue`, then your append logic lives a level higher and you're not 5 indentation levels deep. (nitpick)